src/pokemon.c

```diff
@@ -6941,3 +6942,180 @@ void UpdateDaysPassedSinceFormChange(u16 days)
+{
+    u8 type = gMovesInfo[move].type;
+    u16 species = GetMonData(mon, MON_DATA_SPECIES);
+    u16 heldItem = GetMonData(mon, MON_DATA_HELD_ITEM, 0);
+    u16 heldItemEffect = ItemId_GetHoldEffect(heldItem);
+    u16 ability = GetMonAbility(mon);
+    u8 type1 = gSpeciesInfo[species].types[0];
+    u8 type2 = gSpeciesInfo[species].types[1];
+
+    if (move == MOVE_IVY_CUDGEL
+        && (species == SPECIES_OGERPON_WELLSPRING_MASK || species == SPECIES_OGERPON_WELLSPRING_MASK_TERA 
+        || species == SPECIES_OGERPON_HEARTHFLAME_MASK || species == SPECIES_OGERPON_HEARTHFLAME_MASK_TERA 
+        || species == SPECIES_OGERPON_CORNERSTONE_MASK || species == SPECIES_OGERPON_CORNERSTONE_MASK_TERA))
+    {
+        type = type2;
+    }
+    else if (move == MOVE_STRUGGLE)
+    {
+        return TYPE_NORMAL;
+    }
+    else if (move == MOVE_TERA_BLAST && GetActiveGimmick(battler) == GIMMICK_TERA && gBattleMons[battler].species == species)
+    {
+        return GetMonData(mon, MON_DATA_TERA_TYPE);
+    }
+    else if (move == MOVE_TERA_STARSTORM && species == SPECIES_TERAPAGOS_STELLAR)
+    {
+        return TYPE_STELLAR;
+    }
+    else if (move == MOVE_HIDDEN_POWER)
+    {
+        return CalculateHiddenPowerType(mon);
+    }
+    else if (move == MOVE_AURA_WHEEL && species == SPECIES_MORPEKO_HANGRY)
+    {
+        type = TYPE_DARK;
+    }
+    else if (gMovesInfo[move].effect == EFFECT_CHANGE_TYPE_ON_ITEM && heldItemEffect == gMovesInfo[move].argument)
+    {
+        return ItemId_GetSecondaryId(heldItem);
+    }
+    else if (move == MOVE_NATURAL_GIFT && ItemId_GetPocket(heldItem) == POCKET_BERRIES)
+    {
+        return gNaturalGiftTable[ITEM_TO_BERRY(heldItem)].type;
+    }
+    else if (move == MOVE_RAGING_BULL 
+        && (species == SPECIES_TAUROS_PALDEAN_COMBAT_BREED 
+        || species == SPECIES_TAUROS_PALDEAN_BLAZE_BREED 
+        || species == SPECIES_TAUROS_PALDEAN_AQUA_BREED))
+    {
+        return type2;
+    }        
+    else if (move == MOVE_REVELATION_DANCE)
+    {
+        if (gBattleMons[battler].species != species && type1 != TYPE_MYSTERY)
+        {
+            type = type1;
+        }
+        else if (gBattleMons[battler].species != species && type2 != TYPE_MYSTERY)
+        {
+            type = type2;
+        }
+        else if (GetBattlerTeraType(battler) != TYPE_STELLAR 
+            && (GetActiveGimmick(battler) == GIMMICK_TERA || IsGimmickSelected(battler, GIMMICK_TERA)))
+        {
+            type = GetMonData(mon, MON_DATA_TERA_TYPE);
+        }        
+        else if (gBattleMons[battler].type1 != TYPE_MYSTERY)
+        {
+            type = gBattleMons[battler].type1;
+        }
+        else if (gBattleMons[battler].type2 != TYPE_MYSTERY)
+        {
+            type = gBattleMons[battler].type2;
+        }
+        else if (gBattleMons[battler].type3 != TYPE_MYSTERY)
+        {
+            type = gBattleMons[battler].type3;
+        }
```

Comment:
I would remove the brackets in this block as well. Basically our rule is. If brackets are necessary for at least one part of the if-else block we add them everywhere otherwise if it is just one line in every condition we leave them out.